agent/artifact_uploader.go

```diff
@@ -327,8 +327,10 @@ func (a *ArtifactUploader) upload(ctx context.Context, artifacts []*api.Artifact
 
 				// Update the states of the artifacts in bulk.
 				err := roko.NewRetrier(
-					roko.WithMaxAttempts(10),
-					roko.WithStrategy(roko.Constant(5*time.Second)),
+					// TODO: roko.ExponentialSubsecond(1*time.Second) WithMaxAttempts(10)
+					// Meanwhile, 8 roko.Exponential(2sec) attempts is 1,2,4,8,16,32,64 seconds delay (~2 mins)
+					roko.WithMaxAttempts(8),
+					roko.WithStrategy(roko.Exponential(2*time.Second, 0)),
```

Comment:
```suggestion
				).DoWithContext(ctx, func(r *roko.Retrier) error {
					timeOutCtx := context.WithTimeout(ctx, 500 * time.Millisecond)
					
					if _, err := a.apiClient.UpdateArtifacts(timeOutCtx, a.conf.JobID, statesToUpload); err != nil {
```

You should be able to use `context.WithTimeout` to avoid having to set the timeout on the http client. Just make sure you make a new context and only use it in this request!

PS don't accept the suggestion, it's not going to work because the diffs can't extend to unedited portions of the file.